sympy/concrete/products.py

```diff
@@ -287,7 +287,7 @@ def _eval_product(self, term, limits):
 
                 p1, p2 = p.as_coeff_Add()
```

Comment:
What you have is fine but this is a good opportunity to point out that list comprehensions are good in this context:

```
p1, p2 = [self._eval_product(i, (k, a, n)) for i in p.as_coeff_Add()]
```